functions/src/project/delete-project.ts

```diff
@@ -40,3 +41,27 @@ export const onDeleteProjectRepositories: CloudFunction<DocumentSnapshot> = fire
     }
 
   });
+
+export const onDeleteProject: CloudFunction<DocumentSnapshot> = firestore
+  .document('projects/{projectUid}')
+  .onDelete(async (projectSnapshot: DocumentSnapshot, context: EventContext) => {
+
+    try {
+      const project: DocumentData = projectSnapshot.data();
+
+      if (Array.isArray(project.monitors) && project.monitors.length > 0) {
+        const promises: Promise<WriteResult>[] = [];
+
+        for (const monitor of project.monitors) {
+          promises.push(deleteMonitorPings(project.uid, monitor.uid))
+        }
+        await Promise.all(promises);
```

Comment:
I think we can return the `Promise.all`?

Reply:
Some places we do and some we don't. Let me check

Reply:
> Note: If a Node.js background function returns a Promise, Cloud Functions ensures that the Promise is settled before terminating.

I think it is better if we always return a promise. I will try to update other functions later on in another PR, but if this one can be updated in this PR please 🤓

Reply:
This will be removed as we will call deleteProjectPings here